plugins/modules/network_compliance_workflow_manager.py

```diff
@@ -0,0 +1,1340 @@
+
+                # Check if a valid response is received
+                if response.get("response"):
+                    self.log("Received API response from 'get_device_list' for device:{0} response: {1}".format(device_ip, str(response)), "DEBUG")
+                    response = response.get("response")
+                    if not response:
+                        continue
+                    for device_info in response:
+                        if device_info["reachabilityStatus"] == "Reachable":
+                            device_id = response[0]["id"]
+                            mgmt_ip_instance_id_map[device_ip] = device_id
+                else:
+                    # If unable to retrieve device information, log an error message
+                    msg = "Unable to retrieve device information for {0}. Please ensure that the device exists and is reachable.".format(device_ip)
+                    self.log(msg, "ERROR")
+                    self.module.fail_json(msg=msg)
+
+            except Exception as e:
+                # Log an error message if any exception occurs during the process
+                error_message = "Error while fetching device ID for device: '{0}' from Cisco Catalyst Center: {1}".format(device_ip, str(e))
+                self.log(error_message, "ERROR")
+
+        return mgmt_ip_instance_id_map
+
+    def get_device_ids_from_site(self, site_name, site_id):
+        """
+        Retrieves the management IP addresses and their corresponding instance UUIDs of devices associated with a specific site in Cisco Catalyst Center.
+
+        Parameters:
+            site_name (str): The name of the site whose devices' information is to be retrieved.
+            site_id (str): The unique identifier of the site.
+
+        Returns:
+            dict: A dictionary mapping management IP addresses to their instance UUIDs.
+
+        Description:
+            This method queries Cisco Catalyst Center to fetch the list of devices associated with the provided site.
+            It then extracts the management IP addresses and their instance UUIDs from the response.
+            Devices that are not reachable are logged as critical errors, and the function fails.
+            If no reachable devices are found for the specified site, it logs an error message and fails.
+
+        """
+        mgmt_ip_instance_id_map = {}
+
+        site_params = {
+            "site_id": site_id,
+        }
+
+        # Attempt to retrieve device information associated with the site
+        try:
+            response = self.dnac._exec(
+                family="sites",
+                function='get_membership',
+                op_modifies=True,
+                params=site_params,
+            )
+            if response:
+                self.log("Received API response from 'get_membership': {0}".format(str(response)), "DEBUG")
+                response = response['device']
+                # Iterate over the devices in the site membership
```

Comment:
can we log the response?

Reply:
Already logging it!

if response:
                self.log("Received API response from 'get_membership': {0}".format(str(response)), "DEBUG")

Reply:
Do you want to log before the if condition?